### PR TITLE
Cadc 11492 Added affinity specs to reduce GPU usage

### PR DESCRIPTION
**Pull request by yeunga**

### CODE REVIEW

File: deployment/k8s-config/harbor/values-keel-dev.yaml [hunk 1]
```diff
@@ -78,7 +143,13 @@ persistence:
 #***************************************************
 core:
   priorityClassName: uber-user-preempt-high
```
Comment (brianmajor) [inline deployment/k8s-config/harbor/values-keel-dev.yaml:152]: I think this priority class name needs to be added to each component of harbor, not just core.

File: deployment/k8s-config/kustomize/base/arc/arc-sshd-deployment.yaml [hunk 2]
```diff
@@ -15,6 +15,13 @@ spec:
       labels:
         run: arc-sshd
     spec:
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: nvidia.com/gpu.count
+                operator: DoesNotExist
```
Comment (brianmajor) [inline deployment/k8s-config/kustomize/base/arc/arc-sshd-deployment.yaml:24]: priority class name should be added to this spec (and to others that don't have it already)

File: deployment/k8s-config/kustomize/base/arc/cavern-tomcat-deployment.yaml [hunk 3]
```diff
@@ -43,6 +43,7 @@ spec:
           readOnly: true
       securityContext:
         runAsUser: 0
+      priorityClassName: uber-user-preempt-high
       serviceAccountName: skaha
       volumes:
       - name: servops-cert
```
Comment (brianmajor) [inline deployment/k8s-config/kustomize/base/arc/cavern-tomcat-deployment.yaml]: this file is missing the gpu affinity

**Review by brianmajor — CHANGES_REQUESTED** [review]
In general the affinity and priority class names need to be contained in all of these files if not there already.

**Review by brianmajor — CHANGES_REQUESTED** [review]
Didn't you need to update the harbor chart version?  That should be part of this PR because it needs to be released on prod.

File: deployment/k8s-config/harbor/values-keel-dev.yaml (hunk 1 above)
Comment (yeunga) [reply deployment/k8s-config/harbor/values-keel-dev.yaml:152]: Updated each harbor component with both affinity and priority class. The priority class of cadc-harbor-jobservice is not set even though it is in the spec.

File: deployment/k8s-config/kustomize/base/arc/arc-sshd-deployment.yaml (hunk 2 above)
Comment (yeunga) [reply deployment/k8s-config/kustomize/base/arc/arc-sshd-deployment.yaml:24]: Added and verified.

File: deployment/k8s-config/kustomize/base/arc/cavern-tomcat-deployment.yaml (hunk 3 above)
Comment (yeunga) [reply deployment/k8s-config/kustomize/base/arc/cavern-tomcat-deployment.yaml]: Removed all cavern related files and directories as cavern has been superseded by arc.

**Review by brianmajor — APPROVED** [review]

**Pull request merged**